src/sample/View.java

```diff
@@ -68,100 +75,105 @@ private void setupGUI()
-        defaultBehaviors.selectedProperty().addListener(new ChangeListener<Boolean>() {
+        // force the field to be numeric only
+        gridSizeStart.textProperty().addListener(new ChangeListener<String>() {
             @Override
-            public void changed(ObservableValue<? extends Boolean> observable, Boolean oldValue, Boolean newValue) {
-                gridSizeStart.setText("");
+            public void changed(ObservableValue<? extends String> observable, String oldValue,
+                                String newValue) {
+                if (!newValue.matches("\\d*")) {
+                    gridSizeStart.setText(newValue.replaceAll("[^\\d]", ""));
+                }
             }
         });
 
+        // button action to start the actual game
+        submitStart.setOnAction(actionEvent -> prepTheGame(defaultBehaviors, gridSizeStart, startGUI));
+        // listener on the checkbox such that the defaults are applied, the text is cleared
+        // the checkbox is also checked on the game running for whether to apply defaults or not
+        defaultBehaviors.selectedProperty().addListener((observable, oldValue,
+                                                         newValue) -> gridSizeStart.setText(""));
+        // add the elements to the grid pane
         this.gridPane.add(handle,0,0);
         this.gridPane.add(argInquire,0, 1);
         this.gridPane.add(gridSizeStart,0,2);
         this.gridPane.add(defaultBehaviors,1,2);
         this.gridPane.add(submitStart,0,3);
     }
 
-    private void prepTheGame(Text helpText, CheckBox defaults, TextField textInput, Stage startGUI)
+    private void prepTheGame(CheckBox defaults, TextField textInput, Stage startGUI)
     {
         int gridSize;
-
+            // if defaults are checked then apply the default 5x5 size
             if (defaults.isSelected())
             {
                 textInput.clear();
-                gridSize = 5;
+                gridSize = Constants.DEFAULT_GRIDSIZE;
             }
             else {
-                try {
                     gridSize = Integer.parseInt(textInput.getText());
-                } catch (NumberFormatException e) {
-                    helpText.setText("Please enter a number or check 'defaults'");
-                    textInput.clear();
-                    return;
+                    if (gridSize > Constants.MAX_GRIDSIZE)
+                    {
+                        gridSize = Constants.MAX_GRIDSIZE;
+                        alertMsgOnMax();
+                    }
                 }
-            }
+            startGUI.close();
+            // apply the computed grid size to the grid pane used in main which
+            // is the central grid pane resource for setting up the game gui
             Main.gridSizeForGame = gridSize;
-            startGameGUI(gridSize, startGUI, defaults.isSelected(), helpText);
+            // start the actual game gui
+            startGameGUI(gridSize);
         }
```

Comment:
Can you  move all this back a tab, it looks like it's indented to the next level.